src/ct_eval.cpp

```diff
@@ -33,6 +33,45 @@ TypedAST::FunctionLiteral* ct_eval(
 	return ast;
 }
 
+MonoId mono_type_from_ast(TypedAST::TypedAST* ast, TypeChecker& tc);
+TypeFunctionId type_func_from_ast(TypedAST::TypedAST* ast, TypeChecker& tc);
+
+TypedAST::TypedAST* ct_eval(
+    TypedAST::Identifier* ast, TypeChecker& tc, TypedAST::Allocator& alloc) {
+	// This does something very similar to what ct_eval(DeclarationList) does.
+	// Maybe it can be de-duplicated?
+
+	assert(ast);
+	assert(ast->m_declaration);
+
+	// TODO: These two lines are fishy; I copied them from
+	// elsewhere, but I don't think we need to go through
+	// the declaration to get the metatype: this is already
+	// done for us by metacheck().
+	MetaTypeId meta_type =
+	    tc.m_core.m_meta_core.find(ast->m_declaration->m_meta_type);
+	ast->m_meta_type = meta_type;
```

Comment:
All of this is already done in metacheck, yes. But not quite. In metacheck there is no find. The identifier just points to the declaration's meta type... I guess this lines can still be deleted?

Reply:
That makes sense to me. I will merge it as-is, and look into it later today